### PR TITLE
Documented Introduction and Guideline to the Test Concept 

### LINKED ISSUE #71: 📚 Add a dedicated “Testing” section and explain the project test strategy
As contributer i would like to to know how the test strategy of the project is implemented, how to use and extend it.

**Acceptance Criteria**

* New _Test Concept Guide_ is documented here: https://docs.securecodebox.io/docs/contributing/test-concept
* It is documented where to implement/find unit-tests, integration-tests.
* It is documented how to use/start unit-tests, integration-tests with make

** further informations**

Reference to common patterns and best practice strategies like: https://martinfowler.com/articles/practical-test-pyramid.html

** topics to cover** 

- [x] Unit Tests for core components like the operator (Go tests)
  - [x] Operator #250 
- [x] Unit Tests for secureCodeBox modules like sdk, hooks, scanner... (jest an pytest)
  - [x] Scanners #250 
  - [x] Hooks  #258 
- [x] Integration Tests 
  - [x] Scanners #250 
  - [x] Hooks #258 
- [ ] Introduction and Guideline to the Test Concept (reference to fowler) - What do we use (unit, integration), what is important (speed) https://github.com/secureCodeBox/documentation/pull/259

### PR DESCRIPTION
**Pull request by Ilyesbdlala**
closes #71

### CODE REVIEW

**Comment by Ilyesbdlala** [comment]
I am currently debugging why the npm build for our documentation website is currently failing.

File: docs/contributing/test-concept/index.md [hunk 1]
```diff
@@ -9,11 +9,19 @@ sidebar_position: 1
 ---
 
 ## Introduction
-In this section we will go over how the different modules (scanner/hook/operator) of secureCodeBox are tested. 
-We will present how each module implements unit and integration tests.
+To ensure the functionality of our project, we make use of automated tests in our Continuous Integration (CI) pipeline. This allows us to have feedback on whether newly implemented and old functionalities are operating correctly.  
+It's also important that this feedback is fast to not delay our operations. That is why, a good balance between unit tests and integration tests has to be struck.  
+For that, we refer to the Martin Fowler article [here](https://martinfowler.com/articles/practical-test-pyramid.html) that presents the fundamentals on creating a good testing pyramid. In a testing pyramid, tests that require more CPU time are at the top of the pyramid and are few. The opposite happens the lower you go in the pyramid.
```
Comment (the-simmon) [inline docs/contributing/test-concept/index.md:14]: I would remove the _here_ and move the link to _article_.  

> we refer to the Martin Fowler [article](https://martinfowler.com/articles/practical-test-pyramid.html) that presents the fundamentals

File: docs/contributing/test-concept/index.md [hunk 2]
```diff
@@ -9,11 +9,19 @@ sidebar_position: 1
 ---
 
 ## Introduction
-In this section we will go over how the different modules (scanner/hook/operator) of secureCodeBox are tested. 
-We will present how each module implements unit and integration tests.
+To ensure the functionality of our project, we make use of automated tests in our Continuous Integration (CI) pipeline. This allows us to have feedback on whether newly implemented and old functionalities are operating correctly.  
+It's also important that this feedback is fast to not delay our operations. That is why, a good balance between unit tests and integration tests has to be struck.  
+For that, we refer to the Martin Fowler article [here](https://martinfowler.com/articles/practical-test-pyramid.html) that presents the fundamentals on creating a good testing pyramid. In a testing pyramid, tests that require more CPU time are at the top of the pyramid and are few. The opposite happens the lower you go in the pyramid.
+We present here how the testing pyramid looks in our case:
```
Comment (the-simmon) [inline docs/contributing/test-concept/index.md:15]: This sounds a bit weird. I would just write: _This is the testing pyramid for the SecureCodeBox_

File: docs/contributing/test-concept/index.md [hunk 3]
```diff
@@ -9,11 +9,19 @@ sidebar_position: 1
...
-We will present how each module implements unit and integration tests.
+To ensure the functionality of our project, we make use of automated tests in our Continuous Integration (CI) pipeline. This allows us to have feedback on whether newly implemented and old functionalities are operating correctly.  
+It's also important that this feedback is fast to not delay our operations. That is why, a good balance between unit tests and integration tests has to be struck.  
+For that, we refer to the Martin Fowler article [here](https://martinfowler.com/articles/practical-test-pyramid.html) that presents the fundamentals on creating a good testing pyramid. In a testing pyramid, tests that require more CPU time are at the top of the pyramid and are few. The opposite happens the lower you go in the pyramid.
+We present here how the testing pyramid looks in our case:
+
+![scb-testing-structure](/img/docs/testing/scb-testing.png)
+
+We run unit tests to our different SCB modules (e.g. Operator, Hook, Scanner ...). These tests are meant to be fast. For our integration test, we run a complete SCB scan in a Kind Cluster and we evaluate the resulted findings.
```
Comment (the-simmon) [inline docs/contributing/test-concept/index.md:19]: _We have unit tests for the different SCB modules_

File: docs/contributing/test-concept/index.md [hunk 4]
```diff
@@ -9,11 +9,19 @@ sidebar_position: 1
...
+We go over here how each module implements unit and integration tests:
```
Comment (the-simmon) [inline docs/contributing/test-concept/index.md:20]: _The following articles describe how each module implements its unit and integration tests:_

**Review by the-simmon — CHANGES_REQUESTED** [review]

**Review by the-simmon — APPROVED** [review]

**Pull request merged**